spec/services/solidus_pay_tomorrow/client/create_order_service_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe SolidusPayTomorrow::Client::CreateOrderService do
+  let(:order) { create(:order_with_line_items, line_items_count: 2) }
+  let(:payment_method) { create(:pt_payment_method) }
+  let(:http_response) { instance_double(HTTParty::Response, parsed_response: success_response, success?: true) }
+  let(:full_name) { Spree::Address::Name.new(order.bill_address.name) }
+  let(:expected_body) do
+    { orderId: order.number,
+      firstName: full_name.first_name,
+      lastName: full_name.last_name,
+      street: order.bill_address.address1,
+      city: order.bill_address.city,
+      zip: order.bill_address.zipcode,
+      state: order.bill_address.state.abbr,
+      email: order.email,
+      returnUrl: 'https://domain.com/webhooks/return',
+      cancelUrl: 'https://domain.com/webhooks/cancel',
+      notifyUrl: 'https://domain.com/webhooks/notify',
+      cellPhone: order.bill_address.phone,
+      loanAmount: order.total.to_i,
+      applicationItems:
+        [{ description: "As seen on TV!", quantity: 1, price: 10.0, sku: "SKU-1" },
+         { description: "As seen on TV!", quantity: 1, price: 10.0, sku: "SKU-2" }] }.to_json
+  end
+
+  before do
+    url = 'https://api-staging.paytomorrow.com/api/application/ecommerce/orders'
+    headers = { 'Authorization': "Bearer #{ENV['ACCESS_TOKEN']}",
```

Comment:
What is this? 

AFAIK `ACCESS_TOKEN` can't be stored on the env variable but we should allow to the client to refresh it anytime we need it.

Reply:
The refresh functionality is out of the scope of this PR. On Friday's call, we decided to do it in another PR when needed.

If not env variable, where should ACCESS_TOKEN be? At this point of time.

Reply:
Added the functionality to always get valid token in this fixup [commit](https://github.com/nebulab/solidus_pay_tomorrow/pull/7/commits/b4915dbdae56e8587c7fc5a2621aa3f945fd3138)

Reply:
One problem I see with this is - all the services' specs that'll inherit from BaseService will have to also stub the access token call. Like - **OrderCreationService**

Since access_token functionality need not be tested in **OrderCreationServiceSpec** , to avoid unnecessary stubbing I have stubbed with _allow_any_instance_of_ which goes against rubocop but keeps the access_token stubbing to a minimal.
Wdyt? @DanielePalombo . Any better way in mind?

Reply:
I can't think of any better way at the moment.
But we have to test that feature somewhere.

Reply:
Yeah, added `base_service_spec.rb` in fixup commit https://github.com/nebulab/solidus_pay_tomorrow/pull/7/commits/3726533cf5f92549ada09fa389eb34c52cfa9ab4